File: ultron/ultron-client/src/main/java/com/github/udoo/ultron/web/AccountController.java
package com.github.udoo.ultron.web;

import com.github.udoo.api.AccountApi;
import org.apache.dubbo.config.annotation.Reference;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * @author dong.yang
 * @data 2019/8/15 10:01
 */
@RestController
@RequestMapping(value = "/account")
public class AccountController {

    @Reference(validation = "true", version = "${dubbo.provider.AccountApi.version}")
    private AccountApi accountApi;

    @Reference(validation = "true", version = "2.0.0")
    private AccountApi accountVersionApi;

    @GetMapping(value = "/{id}.json")
    public Object v1(@PathVariable("id") Integer id) {
        return accountApi.getAccount(id);
    }

    @GetMapping(value = "/v2/{id}.json")
    public Object v2(@PathVariable("id") Integer id) {
        return accountVersionApi.getAccount(id);
    }

}

File: ultron/ultron-server/src/main/java/com/github/udoo/ultron/controller/AccountController.kt
package com.github.udoo.ultron.controller

import com.github.udoo.ultron.common.Result
import com.github.udoo.ultron.common.group.Insert
import com.github.udoo.ultron.model.vo.AccountVO
import com.github.udoo.ultron.service.system.AccountService
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.*

/**
 * @author dong.yang
 * @data 2019/7/23 9:08
 */
@RestController
@RequestMapping("/account")
class AccountController(private val accountService: AccountService) {

    @GetMapping("/{id}.json")
    fun findOne(@PathVariable("id") id: Int): Result<Any> = Result.success(accountService.selectById(id))

    @PostMapping("/save.json")
    fun save(@Validated(value = [Insert::class]) @RequestBody account: AccountVO): Result<Any> {
        accountService.saveOrUpdate(account)
        return Result.success()
    }
}